pkg/aws/aws_client/role.go

```diff
@@ -89,3 +96,202 @@ func (client *AWSClient) ListRoles() ([]types.Role, error) {
 	out, err := client.IamClient.ListRoles(context.TODO(), input)
 	return out.Roles, err
 }
+
+func (client *AWSClient) PolicyAttachedToRole(roleName string, policyArn string) (bool, error) {
+	policies, err := client.ListRoleAttachedPolicies(roleName)
+	if err != nil {
+		return false, err
+	}
+	for _, policy := range policies {
+		if *policy.PolicyArn == policyArn {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (client *AWSClient) ListRoleAttachedPolicies(roleName string) ([]types.AttachedPolicy, error) {
+	policies := []types.AttachedPolicy{}
+	policyLister := iam.ListAttachedRolePoliciesInput{
+		RoleName: &roleName,
+	}
+	policyOut, err := client.IamClient.ListAttachedRolePolicies(context.TODO(), &policyLister)
+	if err != nil {
+		return policies, err
+	}
+	policies = policyOut.AttachedPolicies
+	return policies, nil
```

Comment:
NP we dont need the policies var, we can just return the `policyOut.AttachedPolicies`
```suggestion
	return policyOut.AttachedPolicies, nil
```